modules/@apostrophecms/schema/ui/apos/components/AposInputArray.vue

```diff
@@ -244,6 +245,64 @@ export default {
       if (this.field.max && value.length > this.field.max) {
         return 'max';
       }
+      if (value.length && this.field.fields && this.field.fields.add) {
+        const errorColor = 'var(--a-danger)';
+        const neutralColor = 'var(--a-base-5)';
+
+        const [ uniqueFieldName, uniqueFieldSchema ] = Object.entries(this.field.fields.add).find(([ , subfield ]) => subfield.unique);
+        if (uniqueFieldName) {
+          const duplicates = this.next
+            .map(item => Array.isArray(item[uniqueFieldName])
+              ? item[uniqueFieldName].map(i => i._id).sort().join('|')
+              : item[uniqueFieldName])
+            .filter((item, index, array) => array.indexOf(item) !== index);
+
+          if (duplicates.length) {
+            duplicates.forEach(duplicate => {
+              if (uniqueFieldSchema.type === 'relationship') {
+                nextTick(() => {
+                  duplicate.split('|').forEach(item => {
+                    this.$el.querySelectorAll(`[data-id="${item}"]`).forEach(relation => {
+                      relation.style.borderColor = errorColor;
```

Comment:
Why don't we use classes instead of styles?

Reply:
Done